operators/add_distance.py

```diff
@@ -33,13 +36,27 @@ class VIEW3D_OT_slvs_add_distance(Operator, GenericConstraintOp):
     property_keys = ("value", "align", "flip")
 
     def main(self, context):
-        self.target = context.scene.sketcher.constraints.add_distance(
-            self.entity1,
-            self.entity2,
-            sketch=self.sketch,
-            init=not self.initialized,
-            **self.get_settings(),
-        )
+        if isinstance(self.entity1, SlvsLine2D) and self.entity2 is None:
+            dependencies = self.entity1.dependencies()
+            if (isinstance(dependencies[0], SlvsPoint2D) and 
+                    isinstance(dependencies[1], SlvsPoint2D)):
+                # for loop changes the values of self.entity1 and self.entity2
+                # from a line entity to its endpoints
+                for i in range(0, 2):
+                    self.state_index = i  # set index to access i state_data
```

Comment:
This is problematic, state_index is used to specify the currently active state, to access the data of a specific state you can use the following:
data = self.get_state_data(index)

Reply:
Ok, does this look better?
```python
                for i in range(0, 2):
                    state_data = self.get_state_data(i)
                    state_data["hovered"] = -1
                    state_data["type"] = type(dependencies[i])
                    state_data["is_existing_entity"] = True
                    state_data["entity_index"] = dependencies[i].slvs_index
                self.next_state(context)
```
We still need to make an extra call to self.next_state, because otherwise it will not detect the end of the selection process and expect a second entity from the user.

Reply:
Usually the StatefulOperator class handles the logic to step through the states. Implementation classes shouldn't change that logic and only care about their specific functionality. This is however a local problem with small risk, so i'm ok with the solution if it works.